executor/joiner.go

```diff
@@ -162,6 +196,12 @@ type baseJoiner struct {
 	selected     []bool
 	isNull       []bool
 	maxChunkSize int
+
+	// NOTE:
+	// 1. every columns are used if lUsed/rUsed is nil.
+	// 2. no columns are used if lUsed/rUsed is not nil but the size of lUsed/rUsed is 0.
+	lUsed, rUsed                   []int
```

Comment:
rename `lUsed` and `rUsed` to `lUnpruned` and `rUnpruned`?

// 1. No column will be pruned if both lUnpruned and rUnpruned are nil.
// 2. All the columns will be pruned if lUnpruned and rUnpruned are not nil and the \`len\` of lUnpruned and rUnpruned are 0.

lUnpruned, rUnpruned

Reply:
I think `used` is better for human understading. It's an affirmative voice, and `unpruned` is a negetive voice which makes people think.